applications/RomApplication/tests/structural_dynamic_test_files/MainKratosROM.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import print_function, absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+import KratosMultiphysics
+import KratosMultiphysics.RomApplication as romapp
+
+from KratosMultiphysics.StructuralMechanicsApplication.structural_mechanics_analysis import StructuralMechanicsAnalysis
+from KratosMultiphysics.RomApplication import structural_mechanics_dynamic_implicit_rom_solver
+
+import numpy as np
+import json
+"""
+For user-scripting it is intended that a new class is derived
+from StructuralMechanicsAnalysis to do modifications
+"""
+
+class StructDynamicROM(StructuralMechanicsAnalysis):
+
+    def __init__(self,model,project_parameters):
+        super(StructDynamicROM,self).__init__(model,project_parameters)
+        self.selected_time_step_solution_container = []
+
+    def _CreateSolver(self):        
+        return structural_mechanics_dynamic_implicit_rom_solver.ROMSolver(self.model, self.project_parameters["solver_settings"])
+    
+    def FinalizeSolutionStep(self):
+        super(StructDynamicROM,self).FinalizeSolutionStep()        
+        ArrayOfDisplacements = []
+        if np.isclose(self.time,2) or np.isclose(self.time,4) or np.isclose(self.time,6) or np.isclose(self.time,8) or np.isclose(self.time,10):
+            for node in self._GetSolver().GetComputingModelPart().Nodes:            
+                ArrayOfDisplacements.append(node.GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_X, 0))
+                ArrayOfDisplacements.append(node.GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Y, 0))
+            self.selected_time_step_solution_container.append(ArrayOfDisplacements)
+
+    def ModifyInitialGeometry(self):
+        """Here is the place where the BASIS_ROM and the AUX_ID are imposed to each node"""
+        super(StructDynamicROM,self).ModifyInitialGeometry()
+        
+        computing_model_part = self._solver.GetComputingModelPart()
+
+        with open('NodalModes.json') as f:
+            data=json.load(f)
+            counter = 0
+            rom_dofs=self.project_parameters["solver_settings"]["rom_settings"]["number_of_rom_dofs"].GetInt()
+            Dimensions = len(data['1.0'])  # dimensions are defined here (a vector or a matrix is imported to each node)

```

Comment:
what are u doing here?

what does it mean a tag "1.0" ?? it should be meaningful...

Reply:
i think that the "number_of_rom_dofs" should be written both in the solver_settings etc and in the "data" file

Reply:
In this case, it is checking whether the structure is 2D or 3D. The format of the MainKratos.py to impose the nodal ROM_BASIS can of course change...